src/c/user-settings.js

```diff
@@ -0,0 +1,435 @@
+                  password: m.prop(''),
+                  current_password: m.prop(''),
+                  owner_document: m.prop(user.owner_document),
+                  country_id: m.prop(user.address.country_id),
+                  street: m.prop(user.address.street),
+                  number: m.prop(user.address.number),
+                  city: m.prop(user.address.city),
+                  zipcode: m.prop(user.address.zipcode),
+                  complement: m.prop(user.address.complement),
+                  neighbourhood: m.prop(user.address.neighbourhood),
+                  state: m.prop(user.address.state),
+                  phonenumber: m.prop(user.address.phonenumber),
+                  name: m.prop(user.name)
+              },
+              user_id = args.userId,
+              error = m.prop(''),
+              showEmailForm = h.toggleProp(false, true),
+              countries = m.prop(),
+              states = m.prop(),
+              countriesLoader = postgrest.loader(models.country.getPageOptions()),
+              statesLoader = postgrest.loader(models.state.getPageOptions()),
+              loader = m.prop(true),
+              onSubmit = () => {
+                if (fields.email() !== fields.email_confirmation()){
+                  error('Confirmação de email está incorreta.');
+                }
+                else{
+                  updateUserData(user_id);
+                  m.redraw();
+                }
+
+                  return false;
+              },
+              setCsrfToken = (xhr) => {
+                  if (h.authenticityToken()) {
+                      xhr.setRequestHeader('X-CSRF-Token', h.authenticityToken());
+                  }
+                  return;
+              },
+              updateUserData = (user_id) => {
+                  const userData = {
+                      email: fields.email(),
+                      country_id: fields.country_id(),
+                      address_street: fields.street(),
+                      address_number: fields.number(),
+                      address_city: fields.city(),
+                      address_zip_code: fields.zipcode(),
+                      address_complement: fields.complement(),
+                      address_state: fields.state(),
+                      address_neighbourhood: fields.neighbourhood(),
+                      phone_number: fields.phonenumber(),
+                      owner_document: fields.owner_document(),
+                      current_password: fields.current_password(),
+                      password: fields.password(),
+                      name: fields.name()
+                  };
+
+                  return m.request({
+                      method: 'PUT',
+                      url: `/users/${user_id}.json`,
+                      data: {user: userData},
+                      config: setCsrfToken
+                  });
+              };
+
+        countriesLoader.load().then((data) => {
```

Comment:
We can replace this pattern for `countriesLoader.load().then(countries)`. And let's not forget to always catch our requests and show the relevant errors. A simple inlineError component above the submit button should work.